middlewares/middlewares.go

```diff
@@ -73,7 +74,11 @@ func getRemoteAddr(r *http.Request) string {
 	if len(address) > 0 {
 		return address
 	}
+	if splitIndex := strings.Index(r.RemoteAddr, ":"); splitIndex > 0 {
```

Comment:
`strings.Index` returns `-1` to denote nonexistence of a substring. This could still produce a bad result if (for some reason) `r.RemoteAddr` started with a `:`.